pkg/certificate/providers/certmanager/certificate_manager.go

```diff
@@ -89,7 +90,12 @@ func (cm *CertManager) RotateCertificate(cn certificate.CommonName) (certificate
 	oldCert := cm.cache[cn]
 	cm.cache[cn] = newCert
 	cm.cacheLock.Unlock()
-	cm.announcements <- announcements.Announcement{}
+
+	events.GetPubSubInstance().Publish(events.PubSubMessage{
+		AnnouncementType: announcements.CertificateRotated,
+		NewObj:           cn,
+		OldObj:           cn,
+	})
```

Comment:
For logging, one day could be useful to expose them both? or are we expecting a security concern here
```suggestion
		AnnouncementType: announcements.CertificateRotated,
		NewObj:           newCert,
		OldObj:           oldCert,
	})
```